app/src/main/java/com/ntg/user/mvpsample/tasks/TasksFragment.java

```diff
@@ -0,0 +1,227 @@
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.CheckBox;
+import android.widget.CompoundButton;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import com.ntg.user.mvpsample.R;
+import com.ntg.user.mvpsample.add_task.AddTaskActivity;
+import com.ntg.user.mvpsample.data.Task;
+import com.ntg.user.mvpsample.task_details.TaskDetailsActivity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+import static android.support.v4.util.Preconditions.checkNotNull;
+
+/**
+ * A placeholder fragment containing a simple view.
+ */
+public class TasksFragment extends Fragment implements TasksContract.View {
+
+    FloatingActionButton addTaskFab;
+    @BindView(R.id.rvTasks)
+    RecyclerView rvTasks;
+    @BindView(R.id.noTasksTxtView)
+    TextView noTasksTxtView;
+    private TasksContract.Presenter presenter;
+    TasksAdapter tasksAdapter;
+    LinearLayoutManager llm;
+
+    public TasksFragment() {
+    }
+
+    public static TasksFragment newInstance() {
+        return new TasksFragment();
+    }
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        addTaskFab = getActivity().findViewById(R.id.add_task_fab);
+        addTaskFab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                presenter.addTask();
+            }
+        });
+
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_tasks, container, false);
+        ButterKnife.bind(this, view);
+        llm = new LinearLayoutManager(getContext());
+        rvTasks.setLayoutManager(llm);
+        tasksAdapter = new TasksAdapter(new ArrayList<Task>(0), itemListener);
+        rvTasks.setAdapter(tasksAdapter);
+        return view;
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        presenter.start();
+    }
+
+    @Override
+    public void setPresenter(TasksContract.Presenter presenter) {
+        this.presenter = presenter;
+    }
+
+    @Override
+    public void showTasks(List<Task> tasks) {
+        noTasksTxtView.setVisibility(View.GONE);
+        tasksAdapter.replaceData(tasks);
+    }
+
+    @Override
+    public void showNoTasks() {
+        noTasksTxtView.setVisibility(View.VISIBLE);
+    }
+
+    @Override
+    public void showAddNewTaskUI() {
+        Intent intent = new Intent(getContext(), AddTaskActivity.class);
+        startActivity(intent);
+    }
+
+    @Override
+    public void showTaskDetailsUI() {
+        Intent intent = new Intent(getContext(), TaskDetailsActivity.class);
+        startActivity(intent);
+    }
+
+    @Override
+    public void showTaskAsComplete() {
```

Comment:
remove unused methods , try always to make your code clean without commented and unused code. (General comment)